src/main/scala/com/github/unchama/seichiassist/commands/MineStackCommand.scala

```diff
@@ -1,37 +1,96 @@
+import com.github.unchama.targetedeffect.{SequentialEffect, UnfocusedEffect}
 import com.github.unchama.targetedeffect.commandsender.MessageEffect
 import org.bukkit.command.TabExecutor
 
 object MineStackCommand {
-  val executor: TabExecutor = playerCommandBuilder
-    .argumentsParsers(
-      List(
-        Parsers.fromOptionParser(
-          {
-            case "on"  => Some(true)
-            case "off" => Some(false)
-            case _     => None
-          },
-          MessageEffect("/minestack <on | off> … MineStackの対象アイテムを自動収集するか切り替えます")
-        )
+  def executor(
+    implicit ioCanOpenCategorizedMenu: IO CanOpen CategorizedMineStackMenu
+  ): TabExecutor =
+    BranchedExecutor(
+      Map(
+        "on" -> ChildExecutors.setAutoCollectionExecutor(true),
+        "off" -> ChildExecutors.setAutoCollectionExecutor(false),
+        "open" -> ChildExecutors.openCategorizedMineStackMenu
       )
-    )
-    .execution { context =>
-      IO {
-        val newState = context.args.parsed.head.asInstanceOf[Boolean]
-        val sender = context.sender
-        val pd = SeichiAssist.playermap(sender.getUniqueId).settings
-        UnfocusedEffect {
-          pd.autoMineStack = newState
+    ).asNonBlockingTabExecutor()
+
+  object ChildExecutors {
+
+    def setAutoCollectionExecutor(autoMineStack: Boolean): ContextualExecutor =
+      playerCommandBuilder
+        .execution { context =>
+          IO {
+            val sender = context.sender
+            val pd = SeichiAssist.playermap(sender.getUniqueId).settings
+            SequentialEffect(
+              UnfocusedEffect {
+                pd.autoMineStack = autoMineStack
+              },
+              if (autoMineStack)
+                MessageEffect("mineStack自動収集をonにしました。")
+              else
+                MessageEffect("mineStack自動収集をoffにしました。")
+            )
+          }
+        }
+        .build()
+
+    def openCategorizedMineStackMenu(
+      implicit ioCanOpenCategorizedMenu: IO CanOpen CategorizedMineStackMenu
+    ): ContextualExecutor =
+      playerCommandBuilder
+        .argumentsParsers(
+          List(
+            category => {
+              category.toIntOption match {
+                case Some(categoryValue) =>
+                  MineStackObjectCategory.fromSerializedValue(categoryValue - 1) match {
+                    case Some(category) => succeedWith(category)
+                    case None           => failWith("指定されたカテゴリは存在しません。")
+                  }
+                case None => failWith("カテゴリは数字で入力してください。")
+              }
+            },
+            page => {
+              page.toIntOption match {
+                case Some(pageNum) =>
+                  if (pageNum <= 0) {
+                    failWith("ページ数は正の値を指定してください。")
+                  } else {
+                    succeedWith(page)
+                  }
+                case None =>
+                  failWith("ページ数は数字で入力してください。")
+              }
+            }
```

Comment:
[この辺](https://github.com/GiganticMinecraft/SeichiAssist/blob/b5cdd4f47884afbdbc4a8336bb6a1f595840c1a6/src/main/scala/com/github/unchama/contextualexecutor/builder/Parsers.scala#L23)を使って書くとシンプルになると思います

```suggestion
            Parsers.closedRangeInt(1, Int.MaxValue, "ページ数は正の値を指定してください。")
```